### PR TITLE
Add compare and branch JIT Test

### PR DESCRIPTION
**Pull request by r30shah**
Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
FYI @fjeremic

File: test/functional/JIT_Test/src/jit/test/jitt/misc/SanityCompareAndBranch.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   @Test
+   public void testCompareAndBranch() {
+      SanityCompareAndBranch x = new SanityCompareAndBranch();
+      String a1 = "This is Demo String a1";
+      String a2 = "This is Demo String a2";
+      Assert.assertEquals(0, testAddressCompare(a1,a2));
+      String a3 = "This is Demo String a3";
+      String a4 = "This is Demo String a4";
+      Assert.assertEquals(0, testAddressCompare(a3, a4));
+   }
```
Comment (fjeremic) [inline test/functional/JIT_Test/src/jit/test/jitt/misc/SanityCompareAndBranch.java]: We should add a bit of documentation on what exactly this unit test is trying to capture. Things we should do:

1. Explain what issue the test is trying to capture
2. Reference the fix

There is a decent example of this in [1] for a Z bug.

[1] https://github.com/eclipse/openj9/blob/3fb280271d4b3093c611858b62d13f9fba6c6700/test/functional/JIT_Test/src/jit/test/ra/TestOOLSpill31Bit.java#L41-L49

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by pshipton** [comment]
@fjeremic by running `test all` here and in #8324 very close together, you've restarted jenkins again.

**Comment by fjeremic** [comment]
> @fjeremic by running `test all` here and in #8324 very close together, you've restarted jenkins again.

Ok so this is concerning then. I was under the impression that launching `all jdk8,jdk11` on the same PR is causing the issues, not two separate PRs launching `all jdk8`. Although it happened with me here it could have been any other committer launching a build instead of me and the result would have been the same.

I don't think we should expect committers to start coordinating launching of PR builds. IMO the infrastructure issue needs to be resolved at a very high priority as I don't want to advocate launching less testing for a cross-platform change due to flakiness of Jenkins.

Sorry about this again. @AdamBrousseau / @jdekonin FYI.

**Comment by fjeremic** [comment]
For reference this is the issue tracking the fix:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=553326

I've left a comment there to hopefully encourage more action.

**Comment by pshipton** [comment]
> it could have been any other committer launching a build instead of me

correct, but it has to happen within a few minutes of each other, so the odds are low.

> I don't think we should expect committers to start coordinating launching of PR builds

Agreed. Eclipse is hopefully going to provide more resource, which might fix it. We also have #8252 in plan which might fix it. Until these things occur, we need to be careful not to launch too many jobs all at the same time.

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by r30shah** [comment]
Looking at the failure on macOS 
```
Caused: hudson.remoting.ChannelClosedException: Channel "unknown": Remote call on osx1011-x86-1 failed. The channel is closing down or has closed down
```
I do not see any downstream job that shows failure. This seems like infra-related issue so as other failures.

**Comment by pshipton** [comment]
Not sure why these infrastructure issues occurred, but we can re-run the failed platforms.

Jenkins test sanity win32,win,plinuxxl,osx jdk8

**Pull request merged**